Birthday-2024-Project/Scripts/GridLogic.gd

```diff
@@ -1,17 +1,39 @@
 extends Node2D
+class_name GridLogic
+
+const MAX_HEIGHT = 20
+const MAX_WIDTH = 20
 
 #have this passed by the level selector in the future
 @export var debug_setupData : LevelSetup
 
 var availablePieces : Array[Node2D]
 
 #var 2D array of grid occupancies
+var gridSpaces = []
+var freeSpaces = 0
+
 
 # Called when the node enters the scene tree for the first time.
 func _ready():
```

Comment:
I also think it would be good to have static typing for functions
`func _ready() -> void:`